BundleTransport/app/src/main/java/net/discdd/bundletransport/ServerUploadFragment.java

```diff
@@ -145,15 +147,19 @@ private Void connectToServerComplete(Void x) {
         return null;
     }
 
-    private Void connectToServerError(Exception e) {
+    private Void connectToServerError(Exception e, Context context, String transportTarget) {
         requireActivity().runOnUiThread(() -> {
             serverConnnectedStatus.append("Server exchange incomplete with error.\n");
             serverConnnectedStatus.append("Error: " + e.getMessage() + " \n");
             connectServerBtn.setEnabled(true);
+            new Handler(Looper.getMainLooper()).post(() -> {
+                Toast.makeText(context, "Invalid hostname: " + transportTarget, Toast.LENGTH_SHORT).show();
```

Comment:
i don't think you need to pass in the Context here. can't you call getContext()?

do you also need to create a handler? can you call toast directly?